ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -991,10 +991,17 @@ module Winbind = struct
     in
     String.init len (fun _ -> random_char ())
 
-  let build_netbios_name () =
+  let build_netbios_name localhost_name =
     (* Winbind follow https://docs.microsoft.com/en-US/troubleshoot/windows-server/identity/naming-conventions-for-computer-domain-site-ou#domain-names to limit netbios length to 15
      * Compress the hostname if exceed the length *)
-    let hostname = get_localhost_name () in
+    let netbios_sep = '.' in
```

Comment:
Is this name misleading? The dot is the separator in the FQDN and this code extracts the hostname from the FQDN.

It would be better to rely on a library to split up a FQDN. We already have 

* https://ocaml.org/u/36c4fc6563935290671863522bba451e/domain-name/0.4.0/doc/Domain_name/index.html

in our dependencies.